pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,957 +71,765 @@
+                        "checkpoint_io set through both strategy class and plugins, choose one"
                     )
-            elif isinstance(plug, ClusterEnvironment):
-                if cluster_environment is None:
-                    cluster_environment = plug
                 else:
+                    self.checkpoint_io = self._strategy_flag._checkpoint_io
+            if getattr(self._strategy_flag, "cluster_environment", None):
+                if self._cluster_environment_flag:
                     raise MisconfigurationException(
-                        "You can only specify one cluster environment. Found more than 1 cluster environment plugin"
+                        "cluster_environment set through both strategy class and plugins, choose one"
                     )
-            else:
+                else:
+                    self._cluster_environment_flag = getattr(self._strategy_flag, "cluster_environment")
+            # RFC existing accel_conn doesn't handle this, should we add conflict check?
+            # eg: parallel_device is torch.device(cpu) but accelerator=gpu
+            if hasattr(self._strategy_flag, "parallel_devices"):
+                if self._strategy_flag.parallel_devices:
+                    if self._strategy_flag.parallel_devices[0].type == "cpu":
+                        self._accelerator_flag = "cpu"
+                    if self._strategy_flag.parallel_devices[0].type == "cuda":
+                        self._accelerator_flag = "gpu"
+
+        amp_type = amp_type.lower() if isinstance(amp_type, str) else None
+        self._amp_type_flag = AMPType.from_str(amp_type)
+
+        if amp_level is not None and self._amp_type_flag != AMPType.APEX:
+            raise MisconfigurationException(
+                f"You have asked for `amp_level={amp_level!r}` but it's only supported with `amp_backend='apex'`."
+            )
+        self._amp_level_flag = amp_level
+
+    def _device_config_check_and_set_final_flags(self, devices, num_nodes, num_processes, gpus, ipus, tpu_cores):
+        if num_nodes == "auto":
+            self._num_nodes_flag = 1
+        else:
+            self._num_nodes_flag = int(num_nodes) if num_nodes is not None else 1
+
+        self._device_flag = devices
+        # Delete when remove num_processes, gpus, ipus and tpu_cores
+        self._gpus = gpus
+        self._tpu_cores = tpu_cores
+        gpus = device_parser.parse_gpu_ids(gpus)
+        tpu_cores = device_parser.parse_tpu_cores(tpu_cores)
+        deprecated_devices_specific_flag = num_processes or gpus or ipus or tpu_cores
+        if deprecated_devices_specific_flag and deprecated_devices_specific_flag not in (0, "0"):
+            self._mapping_deprecated_devices_specfic_info_to_accelerator_and_device_flag(
+                devices, deprecated_devices_specific_flag, num_processes, gpus, ipus, tpu_cores
+            )
+        # Delete end
```

Comment:
What is this comment supposed to tell me? :)

Reply:
It's for #10410, the gpus, num_tpus will be deprecated, when we remove it, we will delete the code in delete block